File: AES-CBC-PKCS7Padding.java/AES-CBC-PKCS7Padding.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Author:Tea

import hashlib
from Crypto.Cipher import AES
from Crypto.Hash import SHA256
import base64
from urllib import unquote


def main():
    aes = AesCrypter()
    decrypt_data_Tmp = unquote('我是密文')
    print 'UrlDecode:', decrypt_data_Tmp
    decrypt_data = aes.decrypt(decrypt_data_Tmp)
    print decrypt_data
    encrypt_data = '我是明文'
    print aes.encrypt(encrypt_data)


class AesCrypter(object):

    def __init__(self):  # 消息摘要
        key = 'HAHAHAHAHAHAHAHA'
        hash = SHA256.new()
        hash.update(key)
        self.key = hash.digest()
        # self.key = hashlib.sha256(key).digest()
        print 'Self.Key:', self.key.encode('hex')  # 这里相等
        self.iv = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"  # 正确
        # self.iv = chr(0) * 16  # 正确
        # self.iv = "0000000000000000" 错误
        print 'Self.iv:', self.iv.encode('hex')

    def encrypt(self, data):  # 加密
        data = self.pkcs7padding(data)
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        encrypted = cipher.encrypt(data)
        return base64.b64encode(encrypted)

    def decrypt(self, data):  # 解密
        data = base64.b64decode(data)
        print 'Base64Decode:', data.encode('hex')
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        decrypted = cipher.decrypt(data)
        decrypted = self.pkcs7unpadding(decrypted)
        return decrypted

    def pkcs7padding(self, data):
        bs = AES.block_size
        padding = bs - len(data) % bs
        padding_text = chr(padding) * padding
        return data + padding_text

    def pkcs7unpadding(self, data):
        lengt = len(data)
        unpadding = ord(data[lengt - 1])
        return data[0:lengt - unpadding]


if __name__ == '__main__':
    main()

File: AES-CBC-PKCS7Padding.java/AES-CBC-PKCS7Padding.java
import java.security.MessageDigest;
import java.security.spec.AlgorithmParameterSpec;
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;
import java.security.*;

class AESCrypt {
    private final Cipher cipher;
    private final SecretKeySpec key;
    private AlgorithmParameterSpec spec;

    AESCrypt()
            throws Exception {
        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
        MessageDigest localMessageDigest = MessageDigest.getInstance("SHA-256");
        localMessageDigest.update("HAHAHAHAHAHAHAHA".getBytes("UTF-8"));
        byte[] arrayOfByte = new byte[32];  //  私钥
        System.arraycopy(localMessageDigest.digest(), 0, arrayOfByte, 0, arrayOfByte.length);
        this.cipher = Cipher.getInstance("AES/CBC/PKCS7Padding", "BC");
        this.key = new SecretKeySpec(arrayOfByte, "AES");   // 第一个参数为私钥字节数，第二个为加密方式
        this.spec = getIV();    // Iv ,python表示这里需要 "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"
    }

    public String encrypt(String paramString)
            throws Exception {
        this.cipher.init(1, this.key, this.spec);
        return new String(Base64.getEncoder().encode(this.cipher.doFinal(paramString.getBytes("UTF-8"))), "UTF-8");
    }

    public String decrypt(String paramString)
            throws Exception {
        this.cipher.init(2, this.key, this.spec);
        byte[] arrayOfByte = Base64.getDecoder().decode(paramString);
        return new String(this.cipher.doFinal(arrayOfByte), "UTF-8");
    }

    public AlgorithmParameterSpec getIV() {
        byte[] arrayOfByte = new byte[16];
        arrayOfByte[0] = 0;
        arrayOfByte[1] = 0;
        arrayOfByte[2] = 0;
        arrayOfByte[3] = 0;
        arrayOfByte[4] = 0;
        arrayOfByte[5] = 0;
        arrayOfByte[6] = 0;
        arrayOfByte[7] = 0;
        arrayOfByte[8] = 0;
        arrayOfByte[9] = 0;
        arrayOfByte[10] = 0;
        arrayOfByte[11] = 0;
        arrayOfByte[12] = 0;
        arrayOfByte[13] = 0;
        arrayOfByte[14] = 0;
        arrayOfByte[15] = 0;
        return new IvParameterSpec(arrayOfByte);
    }
}

public class Main {

    public static void main(String[] args) throws Exception {
        String data = "我是密文";
        AESCrypt localAESCrypt = new AESCrypt();
        String decrypt_data = localAESCrypt.decrypt(data);
        System.out.println(decrypt_data);
    }
}
